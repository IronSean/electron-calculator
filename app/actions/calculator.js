// @flow
import type { GetState, Dispatch } from '../reducers/types';

export const ADD_TO_TOTAL = 'ADD_TO_TOTAL';
export const MULTIPLY_TOTAL = 'MULTIPLY_TOTAL';
export const SET_TOTAL = 'SET_TOTAL';

export function addToTotal(value: number) {
  return {
    type: ADD_TO_TOTAL,
    payload: value
  };
}

export function multiplyTotal(value: number) {
  return {
    type: MULTIPLY_TOTAL,
    payload: value
  };
}

export function subtractFomTotal(value: number) {
  return (dispatch: Dispatch) => {
    dispatch(addToTotal(-value));
  };
}

export function divideTotal(value: number) {
  return (dispatch: Dispatch) => {
    if (value === 0) {
      return;
    }

    dispatch(multiplyTotal(1 / value));
  };
}

export function setTotal(value: number) {
  return { type: SET_TOTAL, payload: value };
}

export function clearTotal() {
  return (dispatch: Dispatch) => {
    dispatch(setTotal(0));
  };
}
