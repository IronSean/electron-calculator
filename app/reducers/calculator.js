// @flow
import { ADD_TO_TOTAL, MULTIPLY_TOTAL, SET_TOTAL } from '../actions/calculator';
import type { Action } from './types';

export default function calculator(state: number = 0, action: Action) {
  switch (action.type) {
    case ADD_TO_TOTAL:
      return state + action.payload;
    case MULTIPLY_TOTAL:
      return state * action.payload;
    case SET_TOTAL:
      return action.payload;
    default:
      return state;
  }
}
