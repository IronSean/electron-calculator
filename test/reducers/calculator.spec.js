import calculator from '../../app/reducers/calculator';
import {
  ADD_TO_TOTAL,
  MULTIPLY_TOTAL,
  SET_TOTAL
} from '../../app/actions/calculator';

describe('reducers', () => {
  describe('counter', () => {
    it('should handle initial state', () => {
      expect(calculator(undefined, {})).toMatchSnapshot();
    });

    it('should handle ADD_TO_TOTAL', () => {
      expect(
        calculator(1, { type: ADD_TO_TOTAL, payload: 1 })
      ).toMatchSnapshot();
    });

    it('should handle MULTIPLY_TOTAL', () => {
      expect(
        calculator(1, { type: MULTIPLY_TOTAL, payload: 2 })
      ).toMatchSnapshot();
    });

    it('should handle SET_TOTAL', () => {
      expect(calculator(1, { type: SET_TOTAL, payload: 3 })).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(calculator(1, { type: 'unknown' })).toMatchSnapshot();
    });
  });
});
