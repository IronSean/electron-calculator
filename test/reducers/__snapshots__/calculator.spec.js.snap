// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers counter should handle ADD_TO_TOTAL 1`] = `2`;

exports[`reducers counter should handle MULTIPLY_TOTAL 1`] = `2`;

exports[`reducers counter should handle SET_TOTAL 1`] = `3`;

exports[`reducers counter should handle initial state 1`] = `0`;

exports[`reducers counter should handle unknown action type 1`] = `1`;
